pkg/apis/core/validation/validation.go

```diff
@@ -6000,6 +6001,80 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	containerCount := 0
+	containersWithHostProcessSetTrue := []string{}
+	containersWithHostProcessSetFalse := []string{}
+	containersWhichInheritHostProcessTrue := []string{}
+
+	var podHostProcess *bool
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = podSpec.SecurityContext.WindowsOptions.HostProcess
+	}
+
+	hostNetwork := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.HostNetwork == true {
+		hostNetwork = true
+	}
+
+	// Note: Pod-wide WindowsSecurityContextOptions are inherited by containers that do not explicitly set them.
+	for _, c := range append(podSpec.Containers, podSpec.InitContainers...) {
```

Comment:
use `VisitContainersWithPath(podSpec, fieldPath, func(container *api.Container, path *field.Path) bool { ... return true })` to visit all containers, rather than appending Containers and InitContainers